passes/pmgen/peepopt_muxadd.pmg

```diff
@@ -0,0 +1,57 @@
+pattern muxadd
+//
+// Authored by Akash Levy of Silimate, Inc. under ISC license.
+// Transforms add->mux into mux->add:
+// y = s ? (a + b) : a   ===>   y = a + (s ? b : 0)
+//
+
+state <SigSpec> add_a add_b add_y
+
+match add
+	// Select adder
+	select add->type == $add
```

Comment:
```suggestion
	select add->type == $add
	choice <IdString> A {\A, \B}
        define <IdString> B (A == \A ? \B : \A)
        set add_y port(add, \Y)
        set add_a port(add, A)
        set add_b port(add, B)
        set add_a_signed param(add, (A == \A) ? \A_SIGNED : \B_SIGNED))
        set add_a_id A
```

and replace the branch-and-swap pattern further below

Requires new state
```
state <bool> add_a_signed
state <IdString> add_a_id
```

Reply:
This didn't compile... we adjusted this with `state <IdString> add_a_id`

Reply:
Please resolve if this looks ok now